### DISCUSSION TITLE
Does the anchor should be different since mosiac aug make whole box distribution changed?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [ultralytics/yolov5#4255] [Q&A]
Does the anchor should be different since mosiac aug make whole box distribution changed?

**Comment by lucasjinreal** [comment]
Currenly, I only saw the anchor box is absolute and fixed rather than relative to original w,h.

**Reply by glenn-jocher** [reply]
Autoanchor produces anchors on the training data combined with your hyperparameters, so it accounts for --img-size etc. Mosaic is not resizing images, that is the responsibility of the scale hyp.

**Reply by lucasjinreal** [reply]
@glenn-jocher  thank u for your reply. I understand autoanchor generates anchors automatically. But I want try another strategy:
Say I want train it with original image size without specifiying a fixed --img-size, if a batch images with different sizes, I resized it to the largest one. So in this siutation, original anchor based on original dataset's img sizes still works?

Since the img resized, so actually the targets larger in terms of absolute w and h. what do u think?

**Reply by glenn-jocher** [reply]
@jinfagang --rect will dynamically size batches on the short dimension as you describe. In all cases --img will define the long side of a batch. You would have to customize the dataloader to implement your idea, this is not available off the shelf currently.

**Reply by lucasjinreal** [reply]
@glenn-jocher thanks for your suggestion. Do u think if keep anchor ratio but dynamic make it grow with input shape will work from theoretically？

**Reply by glenn-jocher** [reply]
@jinfagang you have to experiment to see what works best for you custom data and model combination.

**Reply by lucasjinreal** [reply]
@glenn-jocher  In my opinion, I think it won't make model converge. since anchor changes every single batch, when inference, it hardly predict exactly location of objects.
